### ISSUE TITLE
💅 Possibly a false positive in handling useArrowFunction

### ISSUE THREAD

**Issue by rchl** [biomejs/biome#4464]
### Environment information

```bash
v1.9.4
```

### Rule name

useArrowFunction

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAFAAbAB1AGcAaQBuACAAfQAgAGYAcgBvAG0AIAAnAEAAbgB1AHgAdAAvAHQAeQBwAGUAcwAnADsACgAKAGUAeABwAG8AcgB0ACAAZABlAGYAYQB1AGwAdAAgADwAUABsAHUAZwBpAG4APgAgAGYAdQBuAGMAdABpAG8AbgAoACkAIAB7AAoAfQA7AAoA&jsx=false

### Expected result

When exported function uses typescript type assertion (I think that's what it's called but not sure, maybe it's just a type cast) then Biome seems to think that it's function expression and suggests converting it to an arrow function. I think it's a false positive.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by rchl** [comment]
I believe this is not a function expression but a function declaration. And the linter error goes away when removing the type cast/assertion which I believe should have nothing to do with that rule.

**Comment by Conaclos** [comment]
> I believe this is not a function expression but a function declaration. And the linter error goes away when removing the type cast/assertion which I believe should have nothing to do with that rule.

There are two kinds of default exports:

1. Default Function declaration exports;
2. Default expression exports.

When you use a cast, we fell in the second one. You can check this in the [TypeScript Playground](https://www.typescriptlang.org/play/?jsx=0#code/KYDwDg9gTgLgBAE2AMwIYFcA28A8AFTdAcwEsA7APjmXTIGMYSIyAKASjgG8AoAXyA):

```ts
export default <Plugin> function() {}
```

is converted to:

```js
export default (function () {});
```

**Comment by rchl** [comment]
Looks like technically you are right.

In practice, let's think whether this rule makes sense in this particular case. It's an exported function, I don't really see the benefit in converting it to an arrow function.

For the reference here is the rule description: https://biomejs.dev/linter/rules/use-arrow-function/

Maybe this should be a special case that doesn't fall within this rule. I'm not sure what would that technically be.

**Comment by ematipico** [comment]
I think we should stay on the technical side for this one. Treating a function expression as it was a function declaration is technically wrong, and if people mistake this case, I am sure they will learn something new. You can check the CST of the playground you shared, or even the [TypeScript](https://ts-ast-viewer.com/#code/KYDwDg9gTgLgBAE2AMwIYFcA28A8AFTdAcwEsA7APjmXTIGMYSIyAKASjgG8AoAXyA) one.

[There are other ways](https://biomejs.dev/playground/?lintRules=all&files.main.ts=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAFAAbAB1AGcAaQBuACAAfQAgAGYAcgBvAG0AIAAnAEAAbgB1AHgAdAAvAHQAeQBwAGUAcwAnADsACgAKAGMAbwBuAHMAdAAgAHAAbAB1AGcAaQBuADoAIABQAGwAdQBnAGkAbgAgAD0AIAAoACkAIAA9AD4AIAB7AH0AOwAKAAoAZQB4AHAAbwByAHQAIABkAGUAZgBhAHUAbAB0ACAAcABsAHUAZwBpAG4AOwAKAA%3D%3D) to export a default function using an arrow function, users should be encouraged to use those ones.

**Comment by Conaclos** [comment]
Moreover, this cast syntax is considered as obsolete by the TypeScript team. Using the new cast syntax makes it clear that the function is a function expression.
I am closing the issue because I think we should keep things as they are here.
